src/app.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
 // Write your code here
+const readline = require('readline');
+const { generateRandomNumber } = require('./generateRandomNumber');
+const { checkIsValidUserInput } = require('./checkIsValidUserInput');
+const { getBullsAndCows } = require('./getBullsAndCows');
+
+const terminal = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+function playGame() {
+  const secretNumber = generateRandomNumber().toString();
+
+  function askGuess() {
+    terminal.question('Enter your number: ', (userInput) => {
+      if (!checkIsValidUserInput(userInput)) {
+        return askGuess();
+      }
+
+      const { bulls } = getBullsAndCows(secretNumber, userInput);
```

Comment:
The code correctly calculates the number of bulls and cows, but it does not provide any feedback to the user. Consider adding a console.log statement to inform the user about the number of bulls and cows after each guess.